### DISCUSSION TITLE
Timezone configuration

### DISCUSSION THREAD

**Discussion by Gandhi11** [ZuinigeRijder/hyundai_kia_connect_monitor#60] [Q&A]
Is there any way to configure the timezone used in the scripts?

Currently the `Last run` date time is set correctly but all the entries are not in my timezone.

<img width="248" alt="image" src="https://github.com/ZuinigeRijder/hyundai_kia_connect_monitor/assets/4001049/5dd44b27-b575-44e1-b550-6e53683a5f73">

When looking at the code, it seems to take the timezone of the car which is in UTC. It doesn't seems possible to change that.

Is there any way to change the timezone when the data is shown in the spreadsheets or csv?

Any help would be appreciated.

**Comment by ZuinigeRijder** [comment]
The timezone is handled in hyundai_kia_connect_api.

For me it shows the correct local date/time, not the UTC timezone for the mentioned values.
I have an IONIQ 5 in Europe. Which car do you have and which region?
Is the date/time and timezone set correctly on the computer where you run the scripts?
Maybe the hyundai_kia_connect_api does not handle the timezone correctly? 
What is vehicle.timezone showing?

**Reply by Gandhi11** [reply]
I have a IONIQ 5 in Canada, UTC-4 currently.

When I run the `debug.py`, the output is saying `timezone=datetime.timezone.utc,` in the `vm.vehicles`

```
         last_updated_at=datetime.datetime
        (
            2024,
             5,
             23,
             10,
             3,
             17,
             tzinfo=datetime.timezone.utc
        ),
         timezone=datetime.timezone.utc,
```

**Comment by ZuinigeRijder** [comment]
For me it is:

````
last_updated_at=datetime.datetime
        (
            2024,
             5,
             22,
             12,
             41,
             24,
             tzinfo=tzfile
            (
                'Europe/Berlin'
            )
        ),
         timezone=tzfile
        (
            'Europe/Berlin'
        ),
````

That is UTC+2, so that is correct for me.

So I assume there is something wrong for the timezone for Canada in hyundai_kia_connect_api.
Note that for Europe, the KiaUvoApiEU.py is used, for Canada KiaUvoApiCA.py. So I expect the timezone is not set correctly for the latter? At least I expect the behavior similar?

Probably best to make an issue in [hyundai_kia_connect_api](https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/issues). And discuss it there further....

Note that in KiaUvoApiEU.py, the default timezone is set. This is not done in KiaUvoApiCA.py.

````
class KiaUvoApiEU(ApiImplType1):
    data_timezone = tz.gettz("Europe/Berlin")
````

But probably also the above is not correct too and should get/convert to the local timezone whenever a datetime is stored, as well for all other regions?

**Comment by Gandhi11** [comment]
Thanks for your help. I created an issue. https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/issues/561
